### PR TITLE
New search URLs

### PR DESCRIPTION
**Pull request by Jeymisson**
#### What problem is this solving?
Resolve more friendly URLS for filters

In order to have a platform more SEO friendly we are changing our searches URL structure when applying filters. This is an workaround that should be resolved when we have a complete solution with collections where each collection will be a filter with a nice structured URL.

#### How should this be manually tested?

https://jey--alssports.myvtex.com

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: node/clients/search.ts [hunk 1]
```diff
@@ -14,6 +14,36 @@ interface AutocompleteArgs {
   searchTerm: string
 }
 
+export interface FieldResponseAPI {
```
Comment (jgfidelis) [inline node/clients/search.ts]: move this type to `node/typings/Catalog.ts`

File: node/clients/search.ts [hunk 2]
```diff
@@ -14,6 +14,36 @@ interface AutocompleteArgs {
...
+  Position: number
+  IsWizard: boolean
+  IsTopMenuLinkActive: boolean
+  IsSideMenuLinkActive: boolean
+  DefaultValue: string | null
+  FieldGroupId: number
+  FieldGroupName: string
+}
+
+export interface FieldTreeResponseAPI{
```
Comment (jgfidelis) [inline node/clients/search.ts]: this type too

File: node/utils/categoryTreeSearch.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// import * as TypeMoq from 'typemoq'
```
Comment (jgfidelis) [inline node/utils/categoryTreeSearch.test.ts]: ?

File: node/utils/CategoryTreeSegmentsFinder.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  private getCategoryTreeRoot = async () => {
+    const categoryTree = await this.clients.search.categories(0)
+    return categoryTree.reduce((acc, categoryTreeNode) => {
+      const categorySlug = searchSlugify(categoryTreeNode.name)
+      const lazyCategoryTreeNode = {
+        id: categoryTreeNode.id,
+        name: categoryTreeNode.name,
+        hasChildren: categoryTreeNode.hasChildren,
+      }
+      return {...acc, [categorySlug]: lazyCategoryTreeNode}
```
Comment (jgfidelis) [inline node/utils/CategoryTreeSegmentsFinder.ts]: modify acc in place and avoid the reduce-spread pattern

File: node/utils/CategoryTreeSegmentsFinder.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        categoryId = Number(childCategoryId)
+      }
+      result.push(childCategoryId)
+    }
+    return result
+  }
+  
+  private findRootCategorySegment = () => {
+    const { segments, categoryTreeRoot } = this
+    for (let i = 0; i < segments.length; i++){
```
Comment (jgfidelis) [inline node/utils/CategoryTreeSegmentsFinder.ts]: y not use `segments.find`? cleaner than a for const for finding something.

Comment (Jeymisson) [reply node/utils/CategoryTreeSegmentsFinder.ts]: I need the to use findIndex, but it is more elegant than the this for.

File: node/typings/Catalog.ts [hunk 6]
```diff
@@ -234,3 +235,33 @@ interface SearchAutocompleteUnit {
   href: string
   criteria: string
 }
+
+interface FieldTreeResponseAPI{
+  Name:	string
```
Comment (jgfidelis) [inline node/typings/Catalog.ts]: formatting issues here

File: node/utils/CategoryTreeSegmentsFinder.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    const segmentsTail = segments.slice(index+1)
+    const categorySegmentsFromChildren = await this.findCategoriesFromChildren(category.id, segmentsTail)
+    return result.concat(categorySegmentsFromChildren)
+  }
+
+  /**
+   * Fetches a category children from vbase or search
+   * clients: {vbase: VBase, search: Search}
+   */
+  protected lazyFetchChildren = async (id: number) => {
```
Comment (jgfidelis) [inline node/utils/CategoryTreeSegmentsFinder.ts]: curiosidade: por que é protegido? essa classe é extendida? muda algo? O.o

**Pull request merged**